unit_testing/test_math.py

```diff
@@ -24,11 +25,29 @@
 # Generate a list of dummy images with single pixel in the middle
 list_im = [
     # test area
-    (dummy_blob(size_arr=(9, 9, 9), debug=DEBUG)),
+    (dummy_blob(size_arr=(9, 9, 9), coordvox=(4, 4, 4), debug=DEBUG)),
     ]
 
 # noinspection 801,PyShadowingNames
 @pytest.mark.parametrize('im', list_im)
 def test_dilate(im):
-    data_dil = math.dilate(im.data, radius=3, shape='disk', dim=0)
-    a=1
+    # cube (only asserting along one dimension for convenience)
+    data_dil = math.dilate(im.data, size=1, shape='cube')
+    assert np.array_equal(data_dil[4, 2:7, 4], np.array([0, 0, 1, 0, 0]))
+    data_dil = math.dilate(im.data, size=2, shape='cube')
+    assert np.array_equal(data_dil[4, 2:7, 4], np.array([0, 1, 1, 0, 0]))
+    data_dil = math.dilate(im.data, size=3, shape='cube')
+    assert np.array_equal(data_dil[4, 2:7, 4], np.array([0, 1, 1, 1, 0]))
+
+    # cube (only asserting along one dimension for convenience)
```

Comment:
Testing matrix math reminds me a lot of [testing HTTP results](https://vcrpy.readthedocs.io/en/latest/) or [web UIs](https://jestjs.io/docs/en/snapshot-testing) -- lots of detail that you don't want to look over but need to know is reliable. It would be helpful if scientific python had snapshot testing.

In the meantime, I think this is fine.

Reply:
i love that idea. especially because right now, my test is not exhaustive (i.e. i am only testing a small portion of the array, to avoid dumping the entire array in the code). if you have suggestions in that direction that would be awesome